### ISSUE TITLE
Villager Tabs Show If They Have No Profession (Might Show If They're A Nitwit Too) And They Also Seem Show Through Blocks

### ISSUE THREAD

**Issue by SmushyTaco** [Andrew6rant/InventoryTabs#70]

**Comment by Andrew6rant** [comment]
Yeah that is part of the reason the update with Entity support is marked as a beta on CurseForge and Modrinth (and pre-release on GitHub).

There currently isn't any code to manage player view for Entities (so even with the config option switched off, you can still access Entity tabs behind you and through walls.

There also isn't anything filtering tabs - I'm not using coordinates like I do for block tabs (since entities move and that would make the tab view flicker and Screens close) so there is only one tab shown per entity type (standing next to five chest minecarts would only show one), and invalid tabs are also shown (like Nitwits or Allays, anything that is instanceof Inventory, InventoryOwner, or InventoryChangedListener).

This will all be fixed in the future, but I am currently on a bit of a modding hiatus - I have a ton of college work that I need to get done, and also I'm still suffering from a recent head injury which makes it a bit hard to stare at the computer screen.
